### DISCUSSION TITLE
Include profile_id in AMS messages for improved convenience

### DISCUSSION THREAD

**Discussion by drewxa** [amzn/ads-advanced-tools-docs#204] [Ideas]
Hi there,
Currently, AMS messages use only advertiser_id and marketplace_id. However, for all API calls, the profile_id is required in the Amazon-Advertising-API-Scope header. This necessitates an additional step of retrieving the profile_id by advertiser_id, which can be inconvenient.

I propose including the profile_id directly in AMS messages. This adjustment would streamline the development process, eliminating the need for an extra step to fetch the profile_id for each API call. This change would enhance convenience and improve the overall efficiency of working with AMS.

For example:
```json
[
{
        "advertiser_id": "AXXXXXXXXXXX",
        "marketplace_id": "ATVPDKIKX0DER",
        "profile_id": "333332222111",  // the proposed field
        "dataset_id": "budget-usage",
        "budget_scope_id": "111122223333333",
        "budget_scope_type": "CAMPAIGN",
        "advertising_product_type": "sp",
        "budget": 9999.0,
        "budget_usage_percentage": 25.05,
        "usage_updated_timestamp": "2023-12-10T12:28:25Z",
        "sent_timestamp": 1702297711813
},
]
```

Thank you for your time, and I look forward to any feedback.

**Comment by elizafarley** [comment]
@drewxa thanks for providing this feedback! I've passed this request along to our Amazon Marketing Stream product team. If we implement this feature, I'll follow up here.

**Comment by drewxa** [comment]
@elizafarley, 
Could you clarify the association between marketplace_id+advertise_id and profile_id? I attempted to locate relevant details in the documentation, but regrettably, I couldn't discern whether the relationship between marketplace_id+advertise_id and profile_id is one-to-one, or if a single marketplace_id+advertise_id pair can be linked to multiple profile_ids?

**Reply by kdwivedi13** [reply]
Thank you for reaching out to us. While we appreciate your interest, I believe you'll find a comprehensive explanation of profile IDs, their relationship with advertisers and marketplaces, in the linked document: https://advertising.amazon.com/API/docs/en-us/reference/2/profiles. Please review information provided, and let us know if you have any further questions.

**Reply by drewxa** [reply]
I've reviewed the link you provided, but unfortunately, I couldn't locate any details regarding the advertiser_id and its association with the profile_id.
